Replace 'attribute' with 'property' for 'indeterminate'

georapbox opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

<!-- ✍️ Summarize your changes in one or two sentences -->

- Replace 'attribute' with 'property' to accurately reflect that `indeterminate` is a JavaScript property (IDL attribute) of the element, not an HTML attribute.
- Fix a broken link in the "See also" section.

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->

It makes more clear that the `indeterminate` is not an HTML attribute so that there are no misunderstandings for the readers.

### Additional details

<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

georapbox committed:
Replace 'attribute' with 'property' to reflect 'indeterminate' is a JavaScript property

- Replace 'attribute' with 'property' to accurately reflect that `indeterminate` is a JavaScript property (IDL attribute) of the element, not an HTML attribute.
- Fix a broken link in the "See also" section.

estelle commented on files/en-us/web/css/_colon_indeterminate/index.md:
```diff
@@ -7,7 +7,7 @@ browser-compat: css.selectors.indeterminate
 {{CSSRef}}
 
-The **`:indeterminate`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) represents any form element whose state is indeterminate, such as checkboxes which have their HTML [`indeterminate`](/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes) attribute set to `true`, radio buttons which are members of a group in which all radio buttons are unch ...
+The **`:indeterminate`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) represents any form element whose state is indeterminate, such as checkboxes which have their [`indeterminate`](/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes) JavaScript property set to `true`, radio buttons which are members of a group in which all radio buttons are ...
```
```suggestion
The **`:indeterminate`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) represents any form element whose state is indeterminate, such as checkboxes that have been set to an [`indeterminate`](/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes) state with JavaScript, radio buttons which are members of a group in which all radio buttons are unchecked, and {{HTMLElement("progress")}} elements with no `value` attribute.
```
While your statement was factual, the indeterminate IDL page doesn't exist (see the non-link on https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement) so we can't link to it so explained how we get to `true` instead of using `true` with no link. Also explained what an indeterminate progress is, instead of using the term people may not understand.

georapbox replied:
Fair points, I applied the suggestion.

georapbox committed:
Apply suggestion

- Clarification on how indeterminate state is set on an element. 
- Explanation on what indeterminate means for progress element.

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

estelle reviewed (approved):
Thanks! 🎉
